Cargo.toml

```diff
@@ -93,6 +93,8 @@ members = [
   "lib/datadog/grok",
   "lib/datadog/search-syntax",
   "lib/datadog/filter",
+  "lib/datadog/filter-test",
+  "lib/vector-datadog-filter",
```

Comment:
Moved the (public) testing pieces into a separate lib. This provides a common set of Datadog Search Syntax queries and pass/fail events for all implementors to assert against.